### DISCUSSION TITLE
How to stop scroll on modal pop-up & keep smooth scroll active?

### DISCUSSION THREAD

**Discussion by kipedia** [darkroomengineering/lenis#292] [Q&A] [answered]
👋
I have a modal pop-up that requires the background to stop scrolling entirely, I'm using the "data-lenis-stop" and "data-lenis-prevent" properties to make this work. However this also stops the smooth scrolling, which I want to keep on the modal.

Any help would be greatly appreciated, thanks!

**Comment by iamlinkus** [comment]
You shouldn't use the data attributes for that purpose.

There are two methods described [here](https://github.com/studio-freight/lenis?tab=readme-ov-file#instance-methods). 

```js
openModal() {
  // Your modal open code
  lenis.stop()
}

closeModal() {
  // Your modal close code
  lenis.start()
}
```

**Comment by Durotan120** [comment]
oke

**Comment by kipedia** [comment]
> You shouldn't use the data attributes for that purpose.
> 
> There are two methods described [here](https://github.com/studio-freight/lenis?tab=readme-ov-file#instance-methods).
> 
> ```js
> openModal() {
>   // Your modal open code
>   lenis.stop()
> }
> 
> closeModal() {
>   // Your modal close code
>   lenis.start()
> }
> ```

Thanks for the reply!

Unfortunately the link is broken. 

I'm using the data-lenis-stop attribute on modal open, and data-lenis-start on modal close, however my modal itself has overflow behaviour, so I'm using data-lenis-prevent to keep scroll active on the modal itself. 
This method works except for the fact that smooth scrolling seems to stop with the data-lenis-stop attribute, but I want smooth scrolling to remain active on scrolling of the modal. Any ideas?

**Comment by cini9** [comment]
I'm having a similar issue with a drawer. I have ReactLenis activated for the root scroll in my NextJS layout. When I open the drawer, I would like the smooth scroll on the root to stop and smooth scroll on the drawer to be active. As it stands, I am able to stop root scrolling via data-lenis-prevent, but the smooth scroll on the drawer does not work. Is this currently possible ?

**Reply by kipedia** [reply]
Please let me know if you find an solution, still looking myself!

**Reply by cini9** [reply]
Nah I haven't found a solution for this yet. In the end I just accepted my fate that the scroll inside of the drawer is not smooth for the time being. But still on the lookout for workarounds or new versions that will allow nested smooth scrolling.

**Reply by clementroche** [reply]
Lenis already support nested scroll, you just need another ReactLenis instance on your drawer 
`<ReactLenis>scrolling content<ReactLenis>` 🤷‍♂️

**Reply by cini9** [reply]
Unfortunately I couldn't get it to work. I will keep trying when I have a moment and @kipedia let me know if you find a way to get it to work too :)

**Reply by henrybabbage** [reply]
Unable to get this to work in a ChakraUI drawer currently, with a nested ReactLenis instanced inside the drawer.

**Reply by VovaVindar** [reply]
Unable to get nested ReactLenis to work

**Reply by clementroche** [reply]
Please share a reproduction link @VovaVindar

**Reply by VovaVindar** [reply]
@clementroche 

Sry, I will share later. But what I've noticed is that if I remove 'root' in my ReactLenis setup in Next.js, I can't scroll at all, regardless of where ReactLenis is placed:
```
<ReactLenis
      options={{
        lerp: LERP_VALUE,
        touchMultiplier: 0,
      }}
    >
      {children}
    </ReactLenis>
```

Am I missing something?

**Reply by clementroche** [reply]
Are you using `lenis` rather than `@studio-freight/lenis` ?

Also can you be sure that your <ReactLenis> component is scrollable, it should have overflow-y: auto; and a defined height.

**Reply by VovaVindar** [reply]
@clementroche 

import { ReactLenis, useLenis } from "lenis/react";

If I add in my globals.css:
```
.lenis {
  overflow-y: auto;
  height: 100vh;
}
```

than smooth scrolling works, but it breaks my scroll triggers

**Reply by clementroche** [reply]
So it's a issue with your ScrollTrigger implementation then, I guess you need to setup the ReactLenis element as scroller for Scrolltrigger

https://gsap.com/docs/v3/Plugins/ScrollTrigger/scroller/

**Reply by VovaVindar** [reply]
@clementroche 

That solves all! So, I need to add the global class for lenis for <ReactLenis> to work without 'root'

Thank you

**Reply by emreguendogdu** [reply]
@clementroche 

Do you know how can I implement this to Framer Motion?

**Reply by clementroche** [reply]
@osmangund https://github.com/darkroomengineering/lenis/blob/main/packages/react/README.md#framer-motion-integration

**Comment by clementroche** [comment, marked as answer]
Check `prevent` option https://github.com/darkroomengineering/lenis/issues/347

[Example using `prevent` option](https://codepen.io/ClementRoche/pen/emONGYN)
[Example using `wrapper` and `content` options](https://codepen.io/ClementRoche/pen/YPKJGEz)

**Reply by kipedia** [reply]
Thanks for your suggestion! If I understand you correctly you're referencing the data-lenis-prevent option?
I already have this option enabled, which allows the modal to be scrolled while the background remains in place. However, it doesn't "prevent" the slow-scroll functionality to be disabled, which is what I'm trying to solve.

**Reply by koppkvn** [reply]
@clementroche why did you mark it as a solution? We still don't know how to keep the smooth scrolling on a modal 🤦‍♂️

**Reply by clementroche** [reply]
Unless I misunderstood the original issue, the `prevent` option is the way to go. Unfortunately, I can't fix issues specifically for you, but I can guide you toward solutions.

Check [this codepen](https://codepen.io/ClementRoche/pen/emONGYN) if you wanna stop body smooth scroll when modal is opened.

Alternatively, if you need smooth scroll to be applied on the modal you could instantiate a `new Lenis` over the modal using the `wrapper` and `content` options.

As always, sharing a reproduction link is extremely helpful for maintainers.

**Reply by kipedia** [reply]
Unfortunately the codepen is exactly my issue. I want the smoothscroll to continue on the modal, I'm curious why the prevent option isn't preventing it from being disabled.

I'm not familiar with the wrapper and content options, I'll look into that.

**Reply by clementroche** [reply]
@kipedia Here is a [codepen](https://codepen.io/ClementRoche/pen/YPKJGEz) using the `wrapper` and `content` options.

**Reply by MrUnknownji** [reply]
but it won't works if we need scroll enabled only inside the dialog box. if we use lenis?.stop() on dialog open and data-lenis-prevent for dialog box area then that dialog area can still be used to scroll entire page which we obviously don't want.

**Comment by clementroche** [comment]
@kipedia please share a reproduction link

**Comment by aimryoui** [comment]
Any other solutions? I got many dialogs, popups, alerts, etc. in my app, it seems not quite right if I need to use `lenis.stop()` in every modal like that, and also need to enable it again.

I'm using Radix as well as Shadcn, after dig into the code base of one, I see that radix uses `react-remove-scroll` plugin to stop page scrolling when a popup for example, is opened.

```js
import { RemoveScroll } from "react-remove-scroll";
...
const ScrollLockWrapper = disableOutsideScroll ? RemoveScroll : React.Fragment;
```

The problem that Lenis someway override this behavior. How to make Lenis respect that?

**Reply by clementroche** [reply]
You certainly need `prevent` option:

> Manually prevent scroll to be smoothed based on elements traversed by events. If true is returned, it will prevent the scroll to be smoothed. Example: `(node) =>  node.classList.contains('cookie-modal')`

**Reply by aimryoui** [reply]
I got you when reference that option, but that not what I need, or at least I don't know how to make it work properly.

In my case, when any mentioned above element are mounted, a `<span>` element with attribute `[data-radix-focus-guard]` appear, that is a unique attribute of a radix-focus-guard that appears every time a dialog, popup, etc. is mounted. So I've tried:

```tsx
prevent: (node) => node.hasAttribute("data-radix-focus-guard")
```

then I realized that `prevent` option just "check" that node on page load, it means if the element was mounted after page load, like right-clicking to open context-menu, nothing happens, the `prevent` is always false.

so I gonna tried with `MutationObserver`, but again, like eventListener one, it will cause many perf problems on the website.

Do you have any ideas, or a small tutorials for someone who need @clementroche ?

**Reply by aimryoui** [reply]
On the other hand, I also tried the [useScrollLock](https://github.com/juliencrn/usehooks-ts/blob/master/packages/usehooks-ts/src/useScrollLock/useScrollLock.ts) hook, but one more time, Lenis overriden the `overflow: hidden` behavior although I've set it to !important.

Unfortunately, Radix doesn't provide any API that can catch if a modal is mounted, so I need to do something with Lenis cuz it's too strict in some edges maybe?

**Reply by clementroche** [reply]
`prevent` is not a check on page load, it gets called on every `wheel` event and Lenis is not overriding any style you applied. You're most likely checking the wrong attribute. You must try with: `prevent: (node) => node.hasAttribute("data-scroll-locked")`

As long as you don't share any reproduction link, we cannot help you.

**Reply by aimryoui** [reply]
thank you so much for replying, I'll try it tomorrow.

**Reply by aimryoui** [reply]
Hi @clementroche, I tried your advice about `[data-scroll-locked]` attribute, genius. But it just work on `<Dialog />` but `<ContextMenu />`.

Please look at this [reproduction](https://codesandbox.io/p/devbox/shadcn-lenis-yg2vrf?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clyzbby0z0006356liwkob7xq%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clyzbby0y0002356lz5exgj0f%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clyzbby0y0004356ltpgltjaf%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clyzbby0y0005356ldqczxca1%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clyzbby0y0002356lz5exgj0f%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyzbby0y0001356lc4gdua84%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clyzdhm8j0002356l0hn7uxpo%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A87%252C%2522startColumn%2522%253A15%252C%2522endLineNumber%2522%253A87%252C%2522endColumn%2522%253A15%257D%255D%252C%2522filepath%2522%253A%2522%252Fcomponents%252Fproviders%252FLayoutProvider.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clyzbby0y0002356lz5exgj0f%2522%252C%2522activeTabId%2522%253A%2522clyzdhm8j0002356l0hn7uxpo%2522%257D%252C%2522clyzbby0y0005356ldqczxca1%2522%253A%257B%2522id%2522%253A%2522clyzbby0y0005356ldqczxca1%2522%252C%2522activeTabId%2522%253A%2522clyzdk0o8001k356lr048xjjr%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clyzbkx1z004e356l32w2lwdv%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522VIRTUAL_MACHINE%2522%252C%2522id%2522%253A%2522clyzdcrxf00v7356l9hhwqrvs%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522SETUP_TASKS%2522%252C%2522id%2522%253A%2522clyzdhmi70006356l2nrfdbt0%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clyzdk0o8001k356lr048xjjr%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clyzbby0y0004356ltpgltjaf%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyzbby0y0003356lz56ua8rv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522clyzbby0y0004356ltpgltjaf%2522%252C%2522activeTabId%2522%253A%2522clyzbby0y0003356lz56ua8rv%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D). Try right clicking anywhere to open the context menu, then try scrolling in and out side of that context menu, it scroll as normal although the `[data-scroll-locked]` attribute is already appear at `body` tag.

Then try opening the `<Dialog />`, it works.

On the other hand. You can toggle lenis with the bottom-left button to see the normal and Lenis' behavior. When Lenis' `options.smoothWheel={false}` the context-menu works.

I also try hard adding the attribute `data-scroll-locked=""` on `context-menu.tsx` at `ContextMenuPrimitive.Content` but hopeless.

Can you tell me if this is Radix bug or Lenis bug?

**Reply by clementroche** [reply]
That's because `pointer-events: none` is applied to body so wheel events are not forwarded. One solution could be to listen attribute change on body using `MutationOberserver` then `lenis.stop` if `data-scroll-locked` is added.

**Reply by aimryoui** [reply]
Thank you for detecting that the problem is from `pointer-events: none`. You're right, `<Dialog />` has a `<DialogOverlay />` that stays in front of the `pointer-events: none` of `<body />` tag, the important here is `<DialogOverlay />` has `pointer-events: auto`.

SO I've solved the problem by adding a `ContextMenuOverlay` (the concept is similar to `<DialogOverlay />`):

```tsx
const ContextMenuOverlay = ({
    className
}: {
    className?: React.HTMLAttributes<HTMLElement>["className"]
}) => (
    <div className={cn(
        "pointer-events-auto fixed inset-0 z-50", // bg-color is transparent
        className
    )} />
)
ContextMenuOverlay.displayName = "ContextMenuOverlay"
```

AND put it inside of `<ContextMenuPortal />` like this:

```tsx
const ContextMenuContent = React.forwardRef<
    React.ElementRef<typeof ContextMenuPrimitive.Content>,
    React.ComponentPropsWithoutRef<typeof ContextMenuPrimitive.Content>
>(({ className, ...props }, ref) => (
    <ContextMenuPortal>
        <>
            <ContextMenuOverlay />
            <ContextMenuPrimitive.Content
                ref={ref}
                className={cn(
                    "...",
                    className
                )}
                {...props}
            />
        </>
    </ContextMenuPortal>
))
```

Finally we can solve it. thank you so much @clementroche

**Reply by Op-Pautu** [reply]
this is a very frustrating problem. spent all day debugging why my navbar dropdown is not disabling scroll (with mousewheel up/down), and it turned out lenis was the problem. the default behaviour should be in lenis, honestly.

anyway the fix is simple. 

``` 
const body = document.body;
  const footer = document.querySelector("footer");
  useEffect(() => {
    const scrollbarWidth =
      window.innerWidth - document.documentElement.clientWidth;

    if (isMenuOpen) {
      // Store current scroll position
      scrollPosition.current = window.scrollY;

      body.classList.add("overflow-hidden", "menu-open");
      body.style.setProperty("--scrollbar-width", `${scrollbarWidth}px`);

      body.setAttribute("data-lenis-prevent", "true");

      if (footer) {
        footer.setAttribute("data-lenis-prevent", "true");
      }
    } else {
      body.classList.remove("overflow-hidden", "menu-open");
      body.removeAttribute("data-lenis-prevent");

      if (footer) {
        footer.removeAttribute("data-lenis-prevent");
      }

      window.scrollTo(0, scrollPosition.current);
    }

    return () => {
      body.classList.remove("overflow-hidden", "menu-open");
      body.removeAttribute("data-lenis-prevent");

      if (footer) {
        footer.removeAttribute("data-lenis-prevent");
      }
    };
  }, [isMenuOpen]);

**Comment by samblongchamps** [comment]
I faced a similar issue with my mobile navigation. I needed to prevent the document from scrolling when it opened. Otherwise, some event listeners inside the nav wouldn't properly register (mostly click events).

Anyway, here's a very basic JS snippet that you can tailor to your own needs:
```typescript
document.addEventListener('DOMContentLoaded', () => {
  const pageWrapper = document.querySelector('html');
  const hamburger = document.querySelector('#hamburger');
  
  let navIsOpen = false;

  hamburger.addEventListener('click', () => {
    navIsOpen = !navIsOpen;
    
    pageWrapper.style.height = navIsOpen ? '100svh' : '';
    pageWrapper.style.overflow = navIsOpen ? 'hidden' : '';
    
    if (navIsOpen){
    	lenis.stop();
    	pageWrapper.setAttribute("data-lenis-prevent", "true");
    }else{
    	lenis.start();
    	pageWrapper.removeAttribute("data-lenis-prevent");
    }
  });
});
```

**Comment by amateur-hacker** [comment]
One approach I found for my use case is to track the state of the data-scroll-locked property on the body element using a MutationObserver.

```tsx
"use client";

import { useEffect, useRef } from "react";

import { cancelFrame, frame, type frameData } from "framer-motion";
import { type LenisRef, ReactLenis } from "lenis/react";

type ReactLenisProviderProps = {
  children: React.ReactNode;
};

const ReactLenisProvider = ({ children }: ReactLenisProviderProps) => {
  const lenisRef = useRef<LenisRef | null>(null);

  useEffect(() => {
    function update(time: typeof frameData) {
      lenisRef.current?.lenis?.raf(time.timestamp);
    }

    frame.update(update, true);

    const observer = new MutationObserver(() => {
      const isScrollLocked = document.body.dataset.scrollLocked === "1";
      const bodyStyle = document.body.style;

      if (lenisRef.current?.lenis) {
        if (isScrollLocked) {
          lenisRef.current.lenis.stop();
          bodyStyle.overflow = "hidden";
        } else {
          lenisRef.current.lenis.start();
          bodyStyle.overflow = "";
        }
      }
    });

    observer.observe(document.body, {
      attributes: true,
      attributeFilter: ["data-scroll-locked"],
    });

    return () => {
      cancelFrame(update);
      observer.disconnect();
    };
  }, []);

  return (
    <ReactLenis
      root
      options={{
        autoRaf: false,
        duration: 1.5,
        lerp: 0.1,
        orientation: "vertical",
        gestureOrientation: "vertical",
      }}
      ref={lenisRef}
    >
      {children}
    </ReactLenis>
  );
};

export { ReactLenisProvider };
```

And also add the data-lenis-prevent attribute to the element. For me, it was the ShadCN components (Dialog, Sheet, and DropdownMenu) content. I added the attribute globally within the respective component itself.

**Comment by feledori** [comment]
Simplest solution is to create a wrapper around the base modal component you are using and listening to an event handler.
Here a simple example using ark-ui's modal component:
```tsx
export function ModalRoot({
  children,
  onOpenChange,
  ...props
}: DialogRootProps) {
  const lenis = useLenis();

  return (
    <DialogRoot
      lazyMount
      unmountOnExit
      preventScroll={false}
      onOpenChange={({ open }) => {
        if (open) {
          lenis?.stop();
        } else {
          lenis?.start();
        }
        onOpenChange?.({ open });
      }}
      {...props}
    >
      {children}
    </DialogRoot>
  );
}
```

Then inside the ModalContent initialise another instance of lenis (non root) to have smooth scroll for the content inside.
```tsx
function ModalContent({
  children,
  positionerClassName,
  contentClassName,
}: ContentProps) {
  return (
      <Dialog.Content>
        <Lenis root={false}>
          {children}
        </Lenis>
      </Dialog.Content>
  );
}
```

This approach assumes you are using a modular component library(you definetly should) which grants you access to the single building blocks of a modal/dialog.

**Comment by Kabin24** [comment]
"use client";

import React from "react";
import { Button } from "./shadcn/button";

type Props = {
  onClose: () => void;
};

const TermsConditions = ({ onClose }: Props) => {
  return (
    <div
      onClick={onClose}
      className="fixed w-screen h-screen inset-0 bg-foreground/60 backdrop-blur-sm flex items-center justify-center z-10"
    >
      <div
        onClick={(e) => e.stopPropagation()}
        data-lenis-prevent
        className="bg-white overflow-y-auto max-h-[90vh] shadow-lg rounded-2xl w-full max-w-4xl "
      >
        {/* Header */}
        <div className="p-6 m-5">
          <h1 className="text-3xl font-semibold pb-2">Terms And Conditions</h1>
          <p className="text-xl text-primary font-medium ">Your Agreement</p>
        </div>

        {/* Content */}
        <div className="px-6   text-sm text-gray-700 space-y-2 m-5    ">
          <p className="">Last Revised: December 16, 2013</p>
          <p>
            Welcome to www.lorem-ipsum.info. This site is provided as a service
            to our visitors and may be used for informational purposes only.
            Because the Terms and Conditions contain legal obligations, please
            read them carefully.
          </p>

          <p className="font-semibold ">1. YOUR AGREEMENT</p>
          <p>
            By using this Site, you agree to be bound by, and to comply with,
            these Terms and Conditions. If you do not agree to these Terms and
            Conditions, please do not use this site.
          </p>

          <p className="text-gray-700">
            <span className="font-bold">PLEASE NOTE:</span> We reserve the
            right, at our sole discretion, to change, modify or otherwise alter
            these Terms and Conditions at any time. Unless otherwise indicated,
            amendments will become effective immediately. Please review these
            Terms and Conditions periodically. Your continued use of the Site
            following the posting of changes and/or modifications will
            constitute your acceptance of the revised Terms and Conditions and
            the reasonableness of these standards for notice of changes. For
            your information, this page was last updated as of the date at the
            top of these terms and conditions.
          </p>

          <p className="font-semibold">2. PRIVACY</p>
          <p>
            Please review our Privacy Policy, which also governs your visit to
            this Site, to understand our practices.which also governs your visit
            to
          </p>

          <p className="font-semibold">3. LINKED SITES</p>
          <p>
            Please review our Privacy Policy, which also governs your visit to
            this Site, to understand our practices.
          </p>
        </div>

        <div className="flex justify-center gap-4  p-4">
          <Button onClick={onClose} className="bg-primary rounded-full px-6">
            Understood
          </Button>
          <Button
            onClick={onClose}
            variant="outline"
            className="rounded-full border-primary  px-6"
          >
            Close
          </Button>
        </div>
      </div>
    </div>
  );
};

export default TermsConditions;
 how to  stop  scrolling
